Lib/fontTools/designspaceLib/__init__.py

```diff
@@ -1222,81 +1135,27 @@ def getAxis(self, name):
-            self.default = flaggedDefaultCandidate
-            self.defaultLoc = self.default.location
-        else:
-            # we have no flagged default candidate
-            # let's use the one from mutator
-            if flaggedDefaultCandidate is None and mutatorDefaultCandidate is not None:
-                # we didn't have a flag, use the one selected by mutator
-                self.default = mutatorDefaultCandidate
-                self.defaultLoc = self.default.location
-        self.default.copyInfo = True
-        # now that we have a default, let's check if the axes are ok
-        for axisObj in self.axes:
-            if axisObj.name not in self.default.location:
-                # extend the location of the neutral master with missing default value for this axis
-                self.default.location[axisObj.name] = axisObj.default
-            else:
-                if axisObj.default == self.default.location.get(axisObj.name):
-                    continue
-                # proposed remedy: change default value in the axisdescriptor to the value of the neutral
-                neutralAxisValue = self.default.location.get(axisObj.name)
-                # make sure this value is between the min and max
-                if axisObj.minimum <= neutralAxisValue <= axisObj.maximum:
-                    # yes we can fix this
-                    axisObj.default = neutralAxisValue
-                    self.logger.info("Note: updating the default value of axis %s to neutral master at %3.3f" % (axisObj.name, neutralAxisValue))
-                # always fit the axis dimensions to the location of the designated neutral
-                elif neutralAxisValue < axisObj.minimum:
-                    axisObj.default = neutralAxisValue
-                    axisObj.minimum = neutralAxisValue
-                elif neutralAxisValue > axisObj.maximum:
-                    axisObj.maximum = neutralAxisValue
-                    axisObj.default = neutralAxisValue
-                else:
-                    # now we're in trouble, can't solve this, alert.
-                    self.logger.info("Warning: mismatched default value for axis %s and neutral master. Master value outside of axis bounds" % (axisObj.name))
-
-    def getMutatorDefaultCandidate(self):
-        # FIXME: original implementation using MutatorMath
-        # masterLocations = [src.location for src in self.sources]
-        # mutatorBias = biasFromLocations(masterLocations, preferOrigin=False)
-        # for src in self.sources:
-        #     if src.location == mutatorBias:
-        #         return src
+                self.default = sourceDescriptor
+                return sourceDescriptor
+        # failing that, well, fail. We don't want to guess any more. 
+        warnings.warn("Can't find a suitable default location in this document.")
```

Comment:
shouldn't this be an error then?

Reply:
or we may simply document that findDefault may return None, and then we don't need to warn here, the user has to handle the None case.